Flaky test: HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded()

https://github.com/apple/servicetalk/pull/1628/checks?check_run_id=2876798025

```
java.lang.AssertionError: Async errors occurred. See suppressed!
...
Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-55-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:271)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:93)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:93)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
		at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
		at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
		at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
		at io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1186)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:773)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:749)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:620)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352)
		at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472)
		at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:957)
		at io.netty.channel.AbstractChannel.close(AbstractChannel.java:244)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.closeChannel(RequestResponseCloseHandler.java:389)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.maybeCloseChannelOnHalfClosed(RequestResponseCloseHandler.java:338)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.channelClosedInbound(RequestResponseCloseHandler.java:219)
		at io.servicetalk.transport.netty.internal.DefaultNettyConnection$NettyToStChannelInboundHandler.userEventTriggered(DefaultNettyConnection.java:647)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:324)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.userEventTriggered(DefaultChannelPipeline.java:1428)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.DefaultChannelPipeline.fireUserEventTriggered(DefaultChannelPipeline.java:913)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.shutdownInput(AbstractEpollChannel.java:526)
		at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:818)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollRdHupReady(AbstractEpollChannel.java:480)
		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:487)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		... 1 more
```

https://github.com/apple/servicetalk/pull/1635/checks?check_run_id=2879808556

```
Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-55-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:271)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:93)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:93)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
		at io.netty.util.concurrent.DefaultPromise.access$200(DefaultPromise.java:35)
		at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:502)
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```

https://github.com/apple/servicetalk/actions/runs/960507106

```
java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:214)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutInvocation.proceed(TimeoutInvocation.java:46)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.stop(TestWorker.java:135)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-55-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:271)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:93)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:93)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
		at io.netty.util.concurrent.DefaultPromise.access$200(DefaultPromise.java:35)
		at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:502)
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```

https://github.com/apple/servicetalk/runs/3184419692

```
java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:211)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutInvocation.proceed(TimeoutInvocation.java:46)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.executeNonConcurrentTasks(ForkJoinPoolHierarchicalTestExecutorService.java:155)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
	Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-203-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:268)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
		at io.netty.util.concurrent.DefaultPromise.access$200(DefaultPromise.java:35)
		at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:502)
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
```

https://github.com/apple/servicetalk/runs/3552974304

```
java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:213)

...

Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-796-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:270)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
		at io.netty.util.concurrent.DefaultPromise.access$200(DefaultPromise.java:35)
		at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:502)
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base/java.lang.Thread.run(Thread.java:829)
```

https://github.com/apple/servicetalk/runs/3563060346

```
java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:213)

...

	Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-796-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:270)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:116)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
		at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
		at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
		at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
		at io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1182)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:773)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:749)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:620)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352)
		at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472)
		at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:957)
		at io.netty.channel.AbstractChannel.close(AbstractChannel.java:244)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.closeChannel(RequestResponseCloseHandler.java:389)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.maybeCloseChannelOnHalfClosed(RequestResponseCloseHandler.java:338)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.channelClosedInbound(RequestResponseCloseHandler.java:219)
		at io.servicetalk.transport.netty.internal.DefaultNettyConnection$NettyToStChannelInboundHandler.userEventTriggered(DefaultNettyConnection.java:688)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:324)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.userEventTriggered(DefaultChannelPipeline.java:1428)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.DefaultChannelPipeline.fireUserEventTriggered(DefaultChannelPipeline.java:913)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.shutdownInput(AbstractEpollChannel.java:526)
		at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:818)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollRdHupReady(AbstractEpollChannel.java:480)
		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:487)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.lang.Thread.run(Thread.java:748)
```

https://github.com/apple/servicetalk/runs/3736884976

```
HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded()

java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:213)
	
...

Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-759-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:270)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:117)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:117)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:552)
		at io.netty.util.concurrent.DefaultPromise.access$200(DefaultPromise.java:35)
		at io.netty.util.concurrent.DefaultPromise$1.run(DefaultPromise.java:502)
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.lang.Thread.run(Thread.java:748)
```

https://github.com/apple/servicetalk/pull/1864/checks?check_run_id=3796489231

```
HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded()
java.lang.AssertionError: Async errors occurred. See suppressed!

java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:213)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
...
	Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-759-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:270)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:117)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:117)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
		at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
		at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
		at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
		at io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1182)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:773)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:749)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:620)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352)
		at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472)
		at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:957)
		at io.netty.channel.AbstractChannel.close(AbstractChannel.java:244)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.closeChannel(RequestResponseCloseHandler.java:389)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.maybeCloseChannelOnHalfClosed(RequestResponseCloseHandler.java:338)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.channelClosedInbound(RequestResponseCloseHandler.java:219)
		at io.servicetalk.transport.netty.internal.DefaultNettyConnection$NettyToStChannelInboundHandler.userEventTriggered(DefaultNettyConnection.java:688)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:324)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.userEventTriggered(DefaultChannelPipeline.java:1428)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.DefaultChannelPipeline.fireUserEventTriggered(DefaultChannelPipeline.java:913)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.shutdownInput(AbstractEpollChannel.java:526)
		at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:818)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollRdHupReady(AbstractEpollChannel.java:480)
		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:487)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base/java.lang.Thread.run(Thread.java:829)
```

https://github.com/apple/servicetalk/pull/1850/checks?check_run_id=3834859595

```
java.lang.AssertionError: Async errors occurred. See suppressed!
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:57)
	at io.servicetalk.test.resources.TestUtils.assertNoAsyncErrors(TestUtils.java:42)
	at io.servicetalk.http.netty.HttpOffloadingTest.serverCloseAsyncGracefullyIsOffloaded(HttpOffloadingTest.java:214)
	
...


	Suppressed: java.lang.AssertionError: onComplete was not offloaded. Thread: io-executor-405-34
		at io.servicetalk.http.netty.HttpOffloadingTest$3.onComplete(HttpOffloadingTest.java:271)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.AfterFinallyCompletable$AfterFinallyCompletableSubscriber.onComplete(AfterFinallyCompletable.java:64)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:117)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.internal.TerminalNotification.terminate(TerminalNotification.java:117)
		at io.servicetalk.concurrent.api.ClosableConcurrentStack.close(ClosableConcurrentStack.java:112)
		at io.servicetalk.concurrent.api.CompletableProcessor.terminate(CompletableProcessor.java:65)
		at io.servicetalk.concurrent.api.CompletableProcessor.onComplete(CompletableProcessor.java:56)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableConcatWithCompletable$ConcatWithSubscriber.onComplete(CompletableConcatWithCompletable.java:69)
		at io.servicetalk.concurrent.api.ContextPreservingCancellableCompletableSubscriber.onComplete(ContextPreservingCancellableCompletableSubscriber.java:39)
		at io.servicetalk.concurrent.api.OnErrorCompleteCompletable$OnErrorCompleteSubscriber.onComplete(OnErrorCompleteCompletable.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.tryToCompleteSubscriber(CompletableMergeSubscriber.java:90)
		at io.servicetalk.concurrent.api.CompletableMergeSubscriber.onComplete(CompletableMergeSubscriber.java:53)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.concurrent.api.ContextPreservingCompletableSubscriber.onComplete(ContextPreservingCompletableSubscriber.java:72)
		at io.servicetalk.transport.netty.internal.NettyFutureCompletable.lambda$connectToSubscriber$1(NettyFutureCompletable.java:52)
		at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
		at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
		at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
		at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
		at io.netty.util.concurrent.DefaultPromise.setSuccess0(DefaultPromise.java:605)
		at io.netty.util.concurrent.DefaultPromise.trySuccess(DefaultPromise.java:104)
		at io.netty.channel.DefaultChannelPromise.trySuccess(DefaultChannelPromise.java:84)
		at io.netty.channel.AbstractChannel$CloseFuture.setClosed(AbstractChannel.java:1182)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.doClose0(AbstractChannel.java:773)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:749)
		at io.netty.channel.AbstractChannel$AbstractUnsafe.close(AbstractChannel.java:620)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.close(DefaultChannelPipeline.java:1352)
		at io.netty.channel.AbstractChannelHandlerContext.invokeClose(AbstractChannelHandlerContext.java:622)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:606)
		at io.netty.channel.AbstractChannelHandlerContext.close(AbstractChannelHandlerContext.java:472)
		at io.netty.channel.DefaultChannelPipeline.close(DefaultChannelPipeline.java:957)
		at io.netty.channel.AbstractChannel.close(AbstractChannel.java:244)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.closeChannel(RequestResponseCloseHandler.java:394)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.maybeCloseChannelOnHalfClosed(RequestResponseCloseHandler.java:343)
		at io.servicetalk.transport.netty.internal.RequestResponseCloseHandler.channelClosedInbound(RequestResponseCloseHandler.java:188)
		at io.servicetalk.transport.netty.internal.DefaultNettyConnection$NettyToStChannelInboundHandler.userEventTriggered(DefaultNettyConnection.java:680)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.AbstractChannelHandlerContext.fireUserEventTriggered(AbstractChannelHandlerContext.java:324)
		at io.netty.channel.DefaultChannelPipeline$HeadContext.userEventTriggered(DefaultChannelPipeline.java:1428)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:346)
		at io.netty.channel.AbstractChannelHandlerContext.invokeUserEventTriggered(AbstractChannelHandlerContext.java:332)
		at io.netty.channel.DefaultChannelPipeline.fireUserEventTriggered(DefaultChannelPipeline.java:913)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.shutdownInput(AbstractEpollChannel.java:526)
		at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:818)
		at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollRdHupReady(AbstractEpollChannel.java:480)
		at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:487)
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ChannelSet.java
+++ b/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/ChannelSet.java
@@ -141,14 +141,14 @@ public final class ChannelSet implements ListenableAsyncCloseable {
                     return;
                 }
 
-                CompositeCloseable closeable = newCompositeCloseable().appendAll(() -> onClose);
+                CompositeCloseable closeable = newCompositeCloseable();
 
                 for (final Channel channel : channelMap.values()) {
                     Attribute<PrivilegedListenableAsyncCloseable> closeableAttribute =
                             channel.attr(CHANNEL_CLOSEABLE_KEY);
                     PrivilegedListenableAsyncCloseable channelCloseable = closeableAttribute.getAndSet(null);
                     if (null != channelCloseable) {
-                        // Upon shutdown of the set, we will close all live channels. If close of individual hannels
+                        // Upon shutdown of the set, we will close all live channels. If close of individual channels
                         // are offloaded, then this would trigger a surge in threads required to offload these closures.
                         // Here we assume that if there is any offloading required, it is done by offloading the
                         // Completable returned by closeAsyncGracefully() hence offloading each channel is not required.
@@ -169,6 +169,7 @@ public final class ChannelSet implements ListenableAsyncCloseable {
                         channel.close();
                     }
                 }
+                closeable.append(() -> onClose);
                 toSource(closeable.closeAsyncGracefully()).subscribe(subscriber);
             }
         };

